File: src/api/dotnet/BoolSort.cs
/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    BoolSort.cs

Abstract:

    Z3 Managed API: Bool Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    /// A Boolean sort.
    /// </summary>
    public class BoolSort : Sort
    {
        #region Internal
        internal BoolSort(Context ctx, IntPtr obj) : base(ctx, obj) { Debug.Assert(ctx != null); }
        internal BoolSort(Context ctx) : base(ctx, Native.Z3_mk_bool_sort(ctx.nCtx)) { Debug.Assert(ctx != null); }
        #endregion
    };
}

File: src/api/java/BoolSort.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    BoolSort.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/

package com.microsoft.z3;

/**
 * A Boolean sort.
 **/
public class BoolSort extends Sort
{
    BoolSort(Context ctx, long obj) { super(ctx, obj); {  }}
    BoolSort(Context ctx) { super(ctx, Native.mkBoolSort(ctx.nCtx())); {  }}
};
